peps/pep-0770.rst

```diff
@@ -179,98 +204,62 @@ Specification
-        Also, the root directory that their paths
-        recorded in the :ref:`Sbom-File <770-spec-sbom-file-field>`
-        :term:`Core Metadata field` are relative to.
-        Defined to be the :term:`project root directory`
-        for a :term:`project source tree` or
-        :term:`source distribution <Source Distribution (or "sdist")>`;
-        and a subdirectory named ``sboms`` of
-        the directory containing the :term:`built metadata`—
-        i.e., the ``.dist-info/sboms`` directory—
-        for a :term:`Built Distribution` or :term:`installed project`.
-
-.. _770-spec-core-metadata:
-
-Core Metadata
--------------
-
-.. _770-spec-sbom-file-field:
-
-Add ``Sbom-File`` field
-~~~~~~~~~~~~~~~~~~~~~~~
-
-The ``Sbom-File`` is a new optional Core Metadata field. Each instance contains a
-string representation of the path to an SBOM document. The path is specified
-relative to the :term:`root SBOM directory` for all project types. It is a
-multi-use field that MAY appear zero or more times and each instance lists the
-path to one such file. Files specified under this field are SBOM documents
-that are distributed with the package.
-
-As `specified by this PEP <#770-spec-project-formats>`__, its value is also
-that file's path relative to the :term:`root SBOM directory` in both installed
-projects and the standardized Distribution Package types.
-
-If an ``Sbom-File`` is listed in a
-:term:`Source Distribution <Source Distribution (or "sdist")>` or
-:term:`Built Distribution`'s Core Metadata:
-
-* That file MUST be included in the :term:`distribution archive` at the
-  specified path relative to the :term:`root SBOM directory`.
-* Installers MUST install the file with the :term:`project` at that same
-  relative path.
-* Inside the :term:`root SBOM directory`, packaging tools MUST reproduce the
-  directory structure under which the source files are located relative to the
-  project root.
-* Path delimiters MUST be the forward slash character (``/``), and parent
-  directory indicators (``..``) MUST NOT be used.
-
-For all newly-uploaded distribution archives that include one or more
-``Sbom-File`` fields in their Core Metadata and declare a ``Metadata-Version``
-of ``2.5`` or higher, PyPI and other indices SHOULD validate that all files
-specified with ``Sbom-File`` are present in the distribution archives.
+    ================= ==========
+    Directory name    PEP
+    ================= ==========
+    ``licenses``      :pep:`639`
+    ``license_files`` N/A (See :ref:`770-backwards-compat`)
+    ``LICENSES``      N/A (See :ref:`770-backwards-compat`)
+    ``sboms``         :pep:`770`
+    ================= ==========
+
+Build backends MUST NOT create subdirectories in the ``.dist-info`` directory
+beyond the names in the registry to avoid collisions with future reserved names.
+
+Build frontends and publishing tools MAY warn users or raise a validation error
+if any ``.dist-info`` subdirectories aren't in the registry.
```

Comment:
I would put this first, before the new registry, as marking all `.dist-info` subdirectories is distinct from the new registry. It may even deserve its own subsection.

Reply:
I am not 100% sure I understood your comment, so can you double-check my changes in https://github.com/python/peps/pull/4283/commits/f0258e8e81db992953ee2424c7af1600b34070cb ?

Reply:
All covered -- thanks!